### PR TITLE
storage: misc logging improvements

### PR DESCRIPTION
**Pull request by tbg**
Got confused by this message in #28995. Prior to this
commit on a merge it looks like the subsuming range is
removing itself, but it's actually the subsumee.

Release note: None

### CODE REVIEW

**Review by a-robinson — APPROVED** [review]
:lgtm: 

> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29000#-:-LKbTTo_0jmSUwhESnEI:bg7igkv)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

___
*[pkg/storage/replica_command.go, line 1050 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/29000#-LKbRvLx3tzlLjVugyod:-LKbRvLx3tzlLjVugyoe:b-vesh5q) ([raw file](https://github.com/cockroachdb/cockroach/blob/e39f6bd012b53c41b555a5d5dab0e436f357d9f6/pkg/storage/replica_command.go#L1050)):*
> ```Go
> 	}
> 
> 	every := log.Every(time.Minute)
> ```

This loop (and the `for len(removeTargets) > 0` loop below) really needs a reasonable backoff on retries. I'll send that out separately, though.

**Comment by tschottdorf** [comment]
TFTR!

@benesch mentioned that RelocateRange needed to be rewritten, so I didn't want to touch it too much. Definitely agree, though.

bors r=a-robinson

**Comment by tschottdorf** [comment]
bors r-

didn't see CI was red.

**Comment by tschottdorf** [comment]
bors r=a-robinson

File: pkg/storage/store.go [hunk 1]
```diff
@@ -2652,6 +2651,10 @@ func (s *Store) removeReplicaImpl(
 		return err
 	}
 
+	// During merges, the context might have the subsuming range, so we explicitly
+	// log the replica to be removed.
+	log.Infof(ctx, "removing replica r%d/%d", rep.RangeID, replicaID)
```
Comment (benesch) [inline pkg/storage/store.go]: Thanks for this! I've been meaning to clean this one up for a while now.

**Comment by a-robinson** [comment]
I assume you want to backport this? I want to backport #29013 and it's easiest to do them together (or at least to do this first).

**Pull request merged**